### ISSUE TITLE
OpenSSL mishandles the Key Usage keyAgreement in ECDH certificates.

### ISSUE THREAD

**Issue by Immortalem** [openssl/openssl#12140]
### Affected Versions
1.1.1g, 1.0.2u

### Setup
OpenSSL has been compiled from source in an alpine docker container. The following excerpt of the used docker file shows the build related commands:
```
RUN ./config --prefix=/build/ --openssldir=/build/ no-async 
RUN make -s && make install_sw -s
```
After building it the respective OpenSSL version and required libraries have been copied into an minimal docker container (from scratch).

The server is later started using the following command:
`openssl s_server -accept 4433 -Verify 100 -cert /cert/inputCerts/rootv3.pem -key /cert/keys/rootv3.pem -CAfile /cert/inputCerts/root.pem -verify_return_error`

All files mentioned in this report (certificates, keys, etc.) have been included in the attached zip archive.

### Issue
OpenSSL allows the usage of ECDH client certificates (key usage set to key agreement only) for the signature verification of the Certificate Verify message. 

### Reproduction
Uses OpenSSL 1.1.1 client on Ubuntu 18.04.
Connect to the started server using the following command:
`openssl s_client -connect localhost:4433 -cert ECROOTv3_CAv3_LEAF_ECv3_KeyAgreement__leaf_certificate1.pem -key eckey_secp256r1_3.pem -CAfile ECROOTv3_CAv3_LEAF_ECv3_KeyAgreement__ca_certificate1.pem -tls1_2`

### Expected Result
OpenSSL should reject the certificate since it may only be used for ECDH key agreement, i.e. it is a static ECDH certificate and may not be used for ECDSA.

### Actual Result
OpenSSL happily accepts the certificate and proceeds with the handshake.

### Attachment
[opensslECKeyAgreement.zip](https://github.com/openssl/openssl/files/4774653/opensslECKeyAgreement.zip)

**Comment by DDvO** [comment]
Bug confirmed. OpenSSL is generally pretty sloppy on checking key usages, at least for SSL/TLS use.
See the new bug report #12721 covering this report and its duplicate #12141.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input

**Comment by DDvO** [comment]
This provides a nice example and test case for the generalized issue #12721.

**Comment by nhorman** [comment]
@DDvO ok, thats fair, but it seems to me that #12721 links back here (even if it is closed), and, as you note covers this use case.  Is there some reason this shouldn't then just be closed as a duplicate of that?

**Comment by DDvO** [comment]
Well, it's not really a duplicate, but a small subset. 
As I wrote, it's an example including a test case, so provides nice detail that should be helpful tackling #12721. 

So in order to minimise the number of open issues. I'd suggest copying the OP there, including the attachment,
and then closing this one.

**Comment by DDvO** [comment]
BTW, can we have a formal relation between issues, such that when closing #12721, this issue is closed automatically?

**Comment by nhorman** [comment]
Unfortunately, github (to the best of my knoweldge) doesn't provide a way to link issues such that both will be closed in parallel.  If you know of one, please let me know, and I'll gladly link them accordingly
